gapic/src/main/com/google/gapid/models/Profile.java

```diff
@@ -361,6 +421,114 @@ protected PerfNode findNodeContaining(Path.Commands commands) {
       });
       return idx < 0 ? null : children.get(idx);
     }
+
+    // Finds all of this node's children, whose range intersects with the given command path.
+    protected List<PerfNode> findAllNodesContaining(Path.Commands commands) {
+      // Find the first child whose range intersects with commands.
+      int start = Collections.binarySearch(children, null, (node, $) ->
+        Paths.compareCommands(node.group.getLink().getToList(), commands.getFromList(), false));
+
+      // Collect all children whose range intersects with commands.
+      List<PerfNode> result = Lists.newArrayList();
+      for (int idx = (start < 0) ? -start - 1 : start; idx < children.size(); idx++) {
+        PerfNode child = children.get(idx);
+        if (Paths.compareCommands(
+            child.group.getLink().getToList(), commands.getToList(), false) <= 0) {
+          result.add(child);
+        } else {
+          break;
+        }
+      }
+      return result;
+    }
+
+    public static interface Value {
+      public String format(Unit unit, boolean estimate);
+    }
+
+    private static class PerfValue implements Value {
+      public final Service.ProfilingData.GpuCounters.Perf perf;
+
+      public PerfValue(Perf perf) {
+        this.perf = perf;
+      }
+
+      @Override
+      public String format(Unit unit, boolean estimate) {
+        if (estimate || perf.getMin() == perf.getMax()) {
+          return unit.format(perf.getEstimate());
+        } else {
+          String minStr = perf.getMin() < 0 ? "?" : unit.format(perf.getMin());
+          String maxStr = perf.getMax() < 0 ? "?" : unit.format(perf.getMax());
+          return minStr + "~" + maxStr;
+        }
+      }
+    }
+
+    private static class ComputedRangeValue implements Value {
+      public final double min;
+      public final double max;
+
+      public ComputedRangeValue(double min, double max) {
+        this.min = min;
+        this.max = max;
+      }
+
+      @Override
+      public String format(Unit unit, boolean estimate) {
+        if (min == max) {
+          return unit.format(min);
+        }
+
+        return unit.format(min) + " - " + unit.format(max);
+      }
+
+      public static ComputedRangeValue aggregate(ComputedRangeValue a, Value b) {
+        if (a == null) {
+          if (b instanceof PerfValue) {
+            double v = ((PerfValue)b).perf.getEstimate();
+            return new ComputedRangeValue(v, v);
+          } else if (b instanceof ComputedRangeValue) {
+            return (ComputedRangeValue)b;
+          }
+        } else if (b instanceof PerfValue) {
+          double v = ((PerfValue)b).perf.getEstimate();
+          return new ComputedRangeValue(Math.min(a.min, v), Math.max(a.max, v));
+        } else if (b instanceof ComputedRangeValue) {
+          ComputedRangeValue v = (ComputedRangeValue)b;
+          return new ComputedRangeValue(Math.min(a.min, v.min), Math.max(a.max, v.max));
+        }
+        return a;
```

Comment:
This function can return null, is that expected?

Reply:
Yes.